src/components/Dashboard.tsx

```diff
@@ -1,46 +1,95 @@
+import { AuthProps } from '@/types/types';
+import dotenv from 'dotenv';
+import { Passage } from '@passageidentity/passage-js';
 
+const getCurrentUserInfo = async () => {
+  dotenv.config();
+  const passageAppId = process.env.NEXT_PUBLIC_PASSAGE_APP_ID;
 
-type DashboardProps = ComponentProps & {
-  logOut: () => void
-};
+  if (!passageAppId) {
+    throw new Error('NEXT_PUBLIC_PASSAGE_APP_ID is not defined in the environment.');
+  }
 
+  const passage = new Passage(passageAppId);
+  const user = passage.getCurrentUser();
+  const userInfo = await user.userInfo();
+  return userInfo
+}
 
-export default function Dashboard({ isAuthorized, data, logOut}: DashboardProps) {
-  const [user, setUser] = useState<UserData | null>(null)
-  const [userInsights, setUserInsights] = useState(insights)
-
+export default function Dashboard({ isAuthorized, userID }: AuthProps) {
+  const [user, setUser] = useState<PassageUserInfo | undefined>(undefined);
+  const [demo, setDemo] = useState(false)
+  // const [userInsights, setUserInsights] = useState(insights)
 
   useEffect(() => {
     if (!isAuthorized) {
       Router.push("/");
-    }      
-  
-    if (data) setUser(data[0])
-    //api call needs to be made to get horoscope
-  }, [isAuthorized]);
-  
+    } 
+    else if (userID !== 'ABrrCENR3M0I6XZ7NLA7gNCY') {
+      getCurrentUserInfo()
+        .then((data) => {
+          setUser(data)
+          checkForUser(userID)
+        })
+        .catch((error) => {
+          console.error('Error fetching user:', error);
+        });
+    } else {
+      setDemo(true)
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const checkForUser = async (userID: string | number) => {
+    const res = await fetch(`/api/getUser?userID=${userID}`)
+    .then((res) => res.json())
+    .then((foundID) => {
+      if (!foundID.length && user) {
+        addNewUser(user)
+      }
+    })
+    .catch((error) => console.log(error))
+  }
+
+  const addNewUser = async (user: PassageUserInfo ) => {
+    const payload = {
+      userID,
+      name: user.user_metadata?.name,
+      email: user.email,
+      birthday: user.user_metadata?.birthday,
+      sign: getZodiacSign(user.user_metadata?.birthday),
+    }
+    const res = await fetch('/api/addUser', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    }).catch(err => console.log(err))
+  };
+
   return (
-    <div className='relative h-full flex flex-col fade-in'>
+    <div className='relative h-full flex flex-col'>
       <div className='mt-10 h-full'>
-        <Logo />
-        <h1 className='mt-7 text-center text-3xl'>Daily Horoscope</h1>
+        <Image className='ml-5' width={300} height={100} style={{ width: '100%', height: 'auto' }} alt="Logo" src={logo} />
+        <h1 className='mt-7 text-center text-3xl'>Daily Horoscope for {user?.user_metadata?.name || ''} {demo && 'Demo'}</h1>
```

Comment:
Obsessed with the addition of the name! 💗

Only issue is that it said `daily horoscope for ` when i logged in and didnt have my name added yet. I dont think it would be a big deal in the future though since name will be required for new users.

I think there might be something that got missed here in a merge conflict. There was a class called fade-in on the div on line 80. I had also changed the image to a separate component to not have to repeat the same info for styling it on different pages. It was called <Logo /> on this branch but will be renaimed to `<CelestialLogo />` after the new PR is merged

Reply:
sweet! Yeah I still have to merge them but i'll make sure those stay!